bitcoin/get_rpc_auth.js

```diff
@@ -0,0 +1,120 @@
+const asyncAuto = require('async/auto');
+const platforms = require('./platforms');
+const {readFile} = require('fs');
+const {returnResult} = require('asyncjs-util');
+
+const fs = {getFile: readFile};
+const {join} = require('path');
+const {keys} = Object;
+const {parse} = require('ini');
+const umbrelPath = '/home/umbrel/umbrel/bitcoin/bitcoin.conf';
+const umbrelUser = 'umbrel';
+
+module.exports = ({os}, cbk) => {
+  return new Promise((resolve, reject) => {
+    return asyncAuto({
+      // Check arguments
+      validate: cbk => {
+        if (!os) {
+          return cbk([400, 'ExpectedOperatingSytemMethodsToGetCookieAuthFile']);
+        }
+      
+        if (!os.homedir) {
+          return cbk([400, 'ExpectedHomedirFunctionToGetCookieAuthFile']);
+        }
+      
+        if (!os.platform) {
+          return cbk([400, 'ExpectedPlatformFunctionToGetCookieAuthFile']);
+        }
+      
+        if (!os.userInfo) {
+          return cbk([400, 'ExpectedUserInfoFunctionToGetCookieAuthFile']);
+        }
+
+        return cbk();
+      },
+
+      //Get Path to cookie file
+      getBitcoinConfPath: ['validate', ({}, cbk) => {
+          // The default directory on Umbrel is not the normal path
+        try {
+          if (os.userInfo().username === umbrelUser) {
```

Comment:
does this return an error sometimes?

Reply:
Not for me, I added it because I saw you added it for getting lnd directory for umbrel, so thought maybe you noticed errors so I put it in too.